### PR TITLE
Read autosubmit pragma and send it in the response

### PR DESCRIPTION
**Pull request by sravfeyn**
https://dimagi-dev.atlassian.net/browse/USH-480

### CODE REVIEW

**Review by ctsims — APPROVED** [review]
Hi Sravan. Good to see your inputs on this, this looks good to me.

File: src/main/java/org/commcare/formplayer/beans/menus/BaseResponseBean.java [hunk 1]
```diff
@@ -8,13 +8,15 @@
...
  * notification - A message String and error boolean to be displayed by frontend
+ * sholdAutoSubmit - A boolean to indicate if the frontend should auto-submit the form
  * clearSession - True if the front end should redirect to the app home
  *  NOTE: clearSession causes all other displayables to be disregarded
  */
 public class BaseResponseBean extends LocationRelevantResponseBean {
     protected NotificationMessage notification;
     protected String title;
     protected boolean clearSession;
+    protected boolean shouldAutoSubmit;
```
Comment (sravfeyn) [inline src/main/java/org/commcare/formplayer/beans/menus/BaseResponseBean.java:19]: I am having trouble having this attributed be returned in the final JSON that the front-end gets. I did `gradle clean, deleted formplayer.jar and built formplayer again. But still, the final JSON didn't have this attribute. Is there any other serializer where I need to add this? 

FWIW just to make sure I compiled correctly, when I renamed `clearSession` var in all palces of the code and the final JSON reflected that change. So, I am not sure what else am I missing to get `shouldAutoSubmit` passed into final formplayer JSON.

Any clues? @ctsims @calellowitz

File: src/main/java/org/commcare/formplayer/beans/menus/BaseResponseBean.java [hunk 2]
```diff
@@ -47,11 +50,15 @@ public void setNotification(NotificationMessage notification) {
...
+    public void shouldAutoSubmit(boolean shouldAutoSubmit) {
+        this.shouldAutoSubmit = shouldAutoSubmit;
+    }
+
+    public boolean isclearSession() {
         return clearSession;
     }
 
-    public void setClearSession(boolean clearSession) {
+    public void setclearSession(boolean clearSession) {
```
Comment (ctsims) [inline src/main/java/org/commcare/formplayer/beans/menus/BaseResponseBean.java:61]: This make break the JSON output, which I believe is defined by these getters and setters

File: src/main/java/org/commcare/formplayer/beans/menus/BaseResponseBean.java [hunk 3]
```diff
@@ -47,11 +50,15 @@ public void setNotification(NotificationMessage notification) {
         this.notification = notification;
     }
 
-    public boolean isClearSession() {
+    public void shouldAutoSubmit(boolean shouldAutoSubmit) {
```
Comment (ctsims) [inline src/main/java/org/commcare/formplayer/beans/menus/BaseResponseBean.java:53]: I believe you need to add a getter (getShouldAutoSubmit) to the class or provide a method annotation for the field to be included in the JSON payload for the bean.

**Review by ctsims — CHANGES_REQUESTED** [review]

**Comment by sravfeyn** [comment]
QA finished, @calellowitz this is good to go now?

https://dimagi-dev.atlassian.net/browse/QA-2165

**Comment by calellowitz** [comment]
Yeah, good from me (and I fixed Clayton's concern). I feel weird approving since two of the commits are mine, but seems good.

**Review by calellowitz — APPROVED** [review]

**Pull request merged**